lib/browser/api/web-contents.ts

```diff
@@ -220,6 +220,16 @@ function parsePageSize (pageSize: string | ElectronInternal.PageSize) {
 let pendingPromise: Promise<any> | undefined;
 WebContents.prototype.printToPDF = async function (options) {
   const margins = checkType(options.margins ?? {}, 'object', 'margins');
+  const pageSize = parsePageSize(options.pageSize ?? 'letter');
+
+  const { top, bottom, left, right } = margins;
+  const validHeight = [top, bottom].every(u => u === undefined || u <= pageSize.paperHeight);
+  const validWidth = [left, right].every(u => u === undefined || u <= pageSize.paperWidth);
```

Comment:
Are 0, negative values valid? If not, do we need to test for >= 1 here?

Reply:
@ckerr yes correct negative values aren't valid, but they're caught in Chromium's helpers so the correct error would be thrown in those cases! Relevant code [here](https://source.chromium.org/chromium/chromium/src/+/main:components/printing/browser/print_to_pdf/pdf_print_utils.cc;l=129-136?q=print_to_pdf::GetPrintPagesParams&sq=package:chromium&type=cs)